logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/conversation/ObserveConversationMembersUseCase.kt

```diff
@@ -18,21 +20,13 @@ class ObserveConversationMembersUseCase(
     private val syncManager: SyncManager
 ) {
 
+    @OptIn(ExperimentalCoroutinesApi::class)
     suspend operator fun invoke(conversationId: ConversationId): Flow<List<MemberDetails>> {
         syncManager.startSyncIfIdle()
-        val selfDetailsFlow = userRepository.observeSelfUser()
-        val selfUser = selfDetailsFlow.first()
-
         return conversationRepository.observeConversationMembers(conversationId).map { members ->
-            members.map {
-                if (it.id == selfUser.id) {
-                    selfDetailsFlow.map(MemberDetails::Self)
-                } else {
-                    userRepository.getKnownUser(it.id).filterNotNull().map { otherUser ->
-                        MemberDetails.Other(
-                            otherUser = otherUser
-                        )
-                    }
+            members.map { member ->
+                userRepository.observeUserInfo(member.id).filterNotNull().map {
+                    MemberDetails(it, member.role)
```

Comment:
honestly I do not see the reason to get the MemberDetails inside the use case, this should be the responsibility of the conversationRepository, since observeUserInfo() is using a UserDao under hood and it looks we do not use it anywhere else, anyway this will not violate multiple repositories depend on each other, also it would gives us more reusaibility in case we need to get MemberDetails in other UseCases, otherwise you would need to create a internal usecase just to get the MemberDetails and inject it here and in the other use case that may be using it to avoid code duplication, but it is your call on this one.